source/cpu/cpu.go

```diff
@@ -276,9 +282,38 @@ func discoverTopology() map[string]string {
 		features["hardware_multithreading"] = strconv.FormatBool(ht)
 	}
 
+	if socketCount, err := getCPUSocketCount(); err != nil {
+		klog.ErrorS(err, "failed to get sockets count")
+	} else {
+		features["socket_count"] = strconv.FormatInt(socketCount, 10)
+	}
+
 	return features
 }
 
+func getCPUSocketCount() (int64, error) {
```

Comment:
Thanks @AhmedGrati for the update.

One note that this `getCPUSocketCount()` looks very similar to `haveThreadSiblings()`. I think an improvement could be to refactor the code by combining these both e.g. directly inside `discoverTopology()`. That could also be a subsequent PR. WDYT?

Reply:
I can do it in the same PR, no problem from my side.

Reply:
Up to you, really. It'd be nice to merge this soon, though 😅

One difference in combining these into one func is probably the error handling. If any (file read) error is encountered then no "topology" features are returned (vs. with the current implementation socket_count can be returned even if smt detection fails). But I think this is pretty theoretical (or not significant) as there should be little to none practical cases where thread_siblings could be read but the physical_package_id not (or vise versa).

Reply:
Got it, let's merge this then, and I'll open another PR for the combination.